### ISSUE TITLE
FFT conv with wrap-around

### ISSUE THREAD

**Issue by LeanderK** [adam-dziedzic/bandlimited-cnns#1]
[Bit offtopic] While researching convolutions via FFT i stumbled upon this library which seems to be very well implemented.
For research I need convolutions via FFT that don't pad the input. It seems trivial but I don't seem to get it right.
I thought I only need to modify:
https://github.com/adam-dziedzic/bandlimited-cnns/blob/39840c979aad22d851bc30043d25e70c49077398/cnns/nnlib/pytorch_layers/conv2D_fft.py#L264
with:
```python
        if out_size_H:
            out_H = out_size_H
        elif out_size or stride_type is StrideType.SPECTRAL:
            out_H = (H - HH + 2 * pad_H) // stride_H + 1
        elif wrap_around: #new
            out_H = H
        else:
            out_H = H - HH + 1 + 2 * pad_H
```
(similarly h)
and https://github.com/adam-dziedzic/bandlimited-cnns/blob/39840c979aad22d851bc30043d25e70c49077398/cnns/nnlib/pytorch_layers/conv2D_fft.py#L290
```python
if wrap_around: #new
   init_H_fft = H + 2 * pad_H
   init_W_fft = W + 2 * pad_W
else:
   (usual init_H_fft, init_W_fft)
```

but this does not equal:
```python
p_x_torch = torch.nn.functional.pad(x_torch, (1,1,1,1), mode='circular')
expect = torch.nn.functional.conv2d(input=p_x_torch, weight=y_torch,
                                            bias=b_torch, padding=0)
```
Thanks for your time!

**Comment by LeanderK** [comment]
Ok, will try.
Yeah, I currently don't have a use for compression, i am just working with the fft-representation for the kernel.

**Comment by LeanderK** [comment]
@adam-dziedzic Thanks! Solved it, the reference padding was wrong. For the future, it was:

`p_x_torch = torch.nn.functional.pad(x_torch, (0,2,0,2), mode='circular')`
(for a 3x3 filter)
